File: src/java/playground/src/Fibonnaci.java
import java.util.ArrayList;
import java.util.List;

public class Fibonnaci {
    public static void main(String[] args) {
        System.out.println("----------------------------------");
        System.out.println("---------- BinarySearch ----------");

        var fiboNumbers = fibonnaci(10);

        fiboNumbers.forEach(it -> System.out.print(it + ", "));


        System.out.println("\n----------------------------------");
    }

    public static List<Integer> fibonnaci(int generateNumber) {
        var list = new ArrayList<Integer>();

        for (int i = 0; i < generateNumber; i++) {
            if (list.isEmpty() || list.size() == 1) {
                list.add(i);
            } else {
                int prevNumber1 = list.get(list.size() - 1);
                int prevNumber2 = list.get(list.size() - 2);
                list.add(prevNumber1 + prevNumber2);
            }
        }

        return list;
    }
}

File: src/kotlin/playground/src/Fibonnaci.kt
fun main() {
    println("----------------------------------")
    println("---------- BinarySearch ----------")

    val fiboNumbers = fibonacci(10)

    fiboNumbers.forEach { println(it) }

    println("----------------------------------")
}

fun fibonacci(generateNumber: Int): List<Int> {
    val list = mutableListOf<Int>()

    for (i in 0 until generateNumber) {
        if (list.isEmpty() || list.size == 1) {
            list.add(i)
        } else {
            val prevNumber1 = list[list.size - 1]
            val prevNumber2 = list[list.size - 2]
            list.add(prevNumber1 + prevNumber2)
        }
    }

    return list
}